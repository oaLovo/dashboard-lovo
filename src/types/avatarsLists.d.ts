export type ListDataProps = {
  avatar?: string
  name: string
  subtitle: string
}

export interface AvatarsListsProps {
  listData: ListDataProps[]
}
