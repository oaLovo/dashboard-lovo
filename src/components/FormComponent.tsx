import styled from 'styled-components'
import { FormComponentProps } from '@/types'
import { StyledButton, StyledInput, StyledP } from '@/components'
import { pxToRem } from '@/utils'

const StyledForm = styled.form`
  display: flex;
  flex-direction: column;
  row-gap: ${pxToRem(16)};
`

function FormComponent(props: FormComponentProps) {
  const { inputs, buttons, message } = props
  return (
    <StyledForm>
      {inputs.map((inputProps, index) => (
        <StyledInput key={index} {...inputProps} />
      ))}
      {buttons.map((buttonProps, index) => (
        <StyledButton key={index} {...buttonProps} />
      ))}
      {message && (
        <StyledP className={message.type === 'error' ? 'error' : 'success'}>
          {message.msg}
        </StyledP>
      )}
    </StyledForm>
  )
}

export default FormComponent
