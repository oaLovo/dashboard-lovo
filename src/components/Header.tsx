import styled from 'styled-components'
import { Link } from 'react-router-dom'

import { Logo } from '@/components'

// MUI
import Avatar from '@mui/material/Avatar'
import Box from '@mui/material/Box'
import Container from '@mui/material/Container'

// UTILS
import { pxToRem } from '@/utils'

const StyledHeader = styled.header`
  background-color: ${(props) => props.theme.appBackground};
  border-bottom: ${pxToRem(1)} solid ${(props) => props.theme.appDefaultStroke};
  margin-bottom: ${pxToRem(37)};
  width: 100%;
`

function Header() {
  return (
    <StyledHeader>
      <Container maxWidth="lg">
        <Box
          sx={{
            alignItems: 'center',
            display: 'flex',
            justifyContent: 'space-between',
            height: pxToRem(64),
          }}
        >
          <Link to="/home">
            <Logo height={30} width={73} />
          </Link>
          <Link to="/perfil">
            <Avatar
              alt="DNC Avatar"
              src="/dnc-avatar.jpg"
              sx={{ width: pxToRem(40), height: pxToRem(40) }}
            />
          </Link>
        </Box>
      </Container>
    </StyledHeader>
  )
}

export default Header
