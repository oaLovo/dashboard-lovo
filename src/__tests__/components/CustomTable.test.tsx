import '@testing-library/jest-dom'
import { CustomTable } from '@/components'
import { CustomTableProps } from '@/types'
import { render } from '@testing-library/react'
import { Theme } from '@/types'
import { ThemeProvider } from 'styled-components'
import { themesList } from '@/resources/themesList'

const defaultProps: CustomTableProps = {
  headers: ['Header 1', 'Header 2', 'Header 3'],
  rows: [
    ['Row 1 Cell 1', 'Row 1 Cell 2', 'Row 1 Cell 3'],
    ['Row 2 Cell 1', 'Row 2 Cell 2', 'Row 2 Cell 3'],
  ],
}

describe('CustomTable Component - Light Theme', () => {
  const renderComponent = (theme: Theme) =>
    render(
      <ThemeProvider theme={theme}>
        <CustomTable {...defaultProps} />
      </ThemeProvider>
    )

  themesList.forEach(({ name, theme }) => {
    describe(`${name}`, () => {
      it('should match the snapshot', () => {
        const { container } = renderComponent(theme)
        expect(container).toMatchSnapshot()
      })

      it('should render headers correctly', () => {
        const { getAllByRole } = renderComponent(theme)
        const headers = getAllByRole('columnheader')
        expect(headers).toHaveLength(defaultProps.headers.length)
        defaultProps.headers.forEach((header, index) => {
          expect(headers[index]).toHaveTextContent(header)
        })
      })

      it('should render rows and cells correctly', () => {
        const { getAllByRole } = renderComponent(theme)
        const rows = getAllByRole('row')
        expect(rows).toHaveLength(defaultProps.rows.length + 1)

        defaultProps.rows.forEach((row, rowIndex) => {
          const cells = rows[rowIndex + 1].querySelectorAll('td')
          expect(cells).toHaveLength(row.length)
          row.forEach((cell, cellIndex) => {
            expect(cells[cellIndex]).toHaveTextContent(String(cell))
          })
        })
      })
    })
  })
})
