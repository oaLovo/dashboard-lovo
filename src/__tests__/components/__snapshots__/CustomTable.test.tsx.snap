// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTable Component - Light Theme Dark Theme should match the snapshot 1`] = `
.c0 {
  overflow-x: auto;
  width: 100%;
}

.c0 table {
  width: 100%;
  border-collapse: collapse;
}

.c0 table .ellipsis {
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 table .ellipsis-sm {
  width: 18.75rem;
}

.c0 table .ellipsis-xs {
  width: 9.375rem;
}

.c0 table th,
.c0 table td {
  height: 3rem;
  padding: 0 0.5rem 0 0;
  text-align: left;
}

.c0 table th:last-child,
.c0 table td:last-child {
  text-align: right;
  padding: 0;
}

.c0 table th {
  color: #89A7CE;
  font-weight: 600;
}

.c0 table tr {
  border-bottom: 0.0625rem solid #21497D;
}

.c0 table tr:last-child {
  border-bottom: none;
}

<div>
  <div
    class="c0"
  >
    <table>
      <thead>
        <tr>
          <th>
            Header 1
          </th>
          <th>
            Header 2
          </th>
          <th>
            Header 3
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Row 1 Cell 1
          </td>
          <td>
            Row 1 Cell 2
          </td>
          <td>
            Row 1 Cell 3
          </td>
        </tr>
        <tr>
          <td>
            Row 2 Cell 1
          </td>
          <td>
            Row 2 Cell 2
          </td>
          <td>
            Row 2 Cell 3
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`CustomTable Component - Light Theme Light Theme should match the snapshot 1`] = `
.c0 {
  overflow-x: auto;
  width: 100%;
}

.c0 table {
  width: 100%;
  border-collapse: collapse;
}

.c0 table .ellipsis {
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 table .ellipsis-sm {
  width: 18.75rem;
}

.c0 table .ellipsis-xs {
  width: 9.375rem;
}

.c0 table th,
.c0 table td {
  height: 3rem;
  padding: 0 0.5rem 0 0;
  text-align: left;
}

.c0 table th:last-child,
.c0 table td:last-child {
  text-align: right;
  padding: 0;
}

.c0 table th {
  color: #828282;
  font-weight: 600;
}

.c0 table tr {
  border-bottom: 0.0625rem solid #E0E0E0;
}

.c0 table tr:last-child {
  border-bottom: none;
}

<div>
  <div
    class="c0"
  >
    <table>
      <thead>
        <tr>
          <th>
            Header 1
          </th>
          <th>
            Header 2
          </th>
          <th>
            Header 3
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Row 1 Cell 1
          </td>
          <td>
            Row 1 Cell 2
          </td>
          <td>
            Row 1 Cell 3
          </td>
        </tr>
        <tr>
          <td>
            Row 2 Cell 1
          </td>
          <td>
            Row 2 Cell 2
          </td>
          <td>
            Row 2 Cell 3
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
